flycheck.el

```diff
@@ -3942,8 +3967,10 @@ Return a list with the contents of the table cell."
                    column 'flycheck-error-list-column-number)
                   (flycheck-error-list-make-cell
                    (symbol-name (flycheck-error-level error)) level-face)
-                  (flycheck-error-list-make-cell
-                   id-str 'flycheck-error-list-id id-str)
+                  ;; Error ID has a button to explain the error
+                  (list id-str
+                        'type 'flycheck-error-list-explain-error
```

Comment:
That might be more intuitive yes.  But it would require to check the existence of the explainer for all errors when constructing the list.  For the moment there's no harm as it will just behave the same as calling `-explain-error-at-point`: if there's no explainer, it will do nothing.